### ISSUE TITLE
Misleading error when typing -exec

### ISSUE THREAD

**Issue by bagage** [sharkdp/fd#221]
I'm so much used to `find` syntax, and so when replacing it with `fd` (which is awesome), I continued to type `-exec` instead of the expected `--exec` (or `-x`). 
But it's only now after few minutes of investigation that I found the reason, the error message being, in my opinion, misleading/unhelpful:

```
$ fd foo -exec rm
Error: 'rm' is not a directory.
$ fd foo -exec rm {} \;
error: Found argument '{}' which wasn't expected, or isn't valid in this context

USAGE:
    fd [FLAGS/OPTIONS] [<pattern>] [<path>]

For more information try --help
```

I know that `-exec` is thus interpreted as `-e=xec` or, more clearly, `--extension xec` which explains why `{}` is not expected here. But since `fd` and `find` share similar syntax, might it make sense to add some userhint `warning: -exec found, did you mean to use --exec instead?` for instance? 
I believe I won't be the last one to fall into this trap :).

**Comment by sharkdp** [comment]
Thank you for the feedback!

I guess we could try to detect that, yes -- but what about the poor people that want to search for `.xec` files? :smile:

**Comment by sharkdp** [comment]
I'm not sure we can distinguish `-e xec` from `-exec` (since we use `clap` to parse command line arguments, and it probably doesn't make that transparent). So we would have to check for that before running `clap`...... or we simply ignore the minuscule probability that someone wants to search for `.xec` files :smile:

Let's implement this.

**Comment by Doxterpepper** [comment]
The problem seems to be that a space is not required between a short arg and the short args value. Clap then doesn't know the difference between -exec as a argument and -e[Value]. I think we could add a validator method to the extension argument to check if there is a space between -e and the value and throw an error appropriately if not.

**Comment by bagage** [comment]
@Doxterpepper I don't think that's a solution -- this is the standard way of handling arguments and `-e value` should behave exactly as `-evalue`. 
At least I suppose that's what people expect to have so I would not break that rule; particularly since this is required only for a very specific use case of `-exec` value. 
If it's impossible to warn user in this single case of `-exec (did you mean --exec?)`, I think it's better to not do anything instead.

**Comment by Doxterpepper** [comment]
@bagage, I'm pretty sure command line arguments are not necessarily standardized. In some cases -e [Value] with a space will behave exactly like -e[Value] without a space but not always. In the case of clap it considers these to be the same but outside of this one library it would not always behave that way. For example if you use GNU grep `grep -G'.*'` gives an error and grep -G '.*' does not.

Since it's not necessarily standardized it would be acceptable behaviour to disallow a space between a short argument and the value that goes with it to avoid ambiguity in arguments like we're seeing here.

**Comment by sharkdp** [comment]
@Doxterpepper I actually think allowing `-e[Value]` is quite standard. I tend to use it a lot (e.g. `fd -tf`, `fd -d1`). GNU grep also supports this style, e.g. `grep -C3`. `-G` is a flag and not an option that takes a value. That's probably why it doesn't work in your example.

**Comment by Doxterpepper** [comment]
@sharkdp, It doesn't seem to matter because we can't do what I'd like using clap. #226 is a valid solution, I was just trying to find something simpler.

**Comment by sharkdp** [comment]
Closed via #226.

**Comment by sharkdp** [comment]
Fix released in v6.3.0
